### PR TITLE
Cart and wishlist management

### PR DESCRIPTION
**Pull request by omkarpatke**
# Cart and wishlist management

## functionality:
- add to cart
- remove from cart
- add to wishlist
- remove from wishlist
- order summary

**Checks:**
- ProductListing.jsx
- Cart.jsx
- Wishlist.jsx

### CODE REVIEW

File: src/pages/Cart/Cart.jsx [hunk 1]
```diff
@@ -1,76 +1,51 @@
 import React from 'react';
+import { removeFromCartlist } from '../../api-calls/api-calls';
+import { useCartlist } from '../../context/cart-context';
 import './Cart.css';
 
 export default function Cart() {
+    let {cartState , cartDispatch} = useCartlist([]);
+    let cartlist;
+    console.log(cartlist)
```
Comment (Sahil25061999) [inline src/pages/Cart/Cart.jsx]: remove console.log if not needed.

**Review by Sahil25061999 — APPROVED** [review]
code looks good, the functionlaity is working well, just a suggestion is that if a product is added to wishlist, you can disable the wishlist button for the product, as it doesnt make sense to add the same product in the wishlist.

File: src/pages/Cart/Cart.jsx (hunk 1 above)
Comment (omkarpatke) [reply src/pages/Cart/Cart.jsx]: yes, remove it when it's done.

File: src/pages/Cart/Cart.jsx [hunk 2]
```diff
@@ -1,103 +1,133 @@
...
+        if(cartState.type === 'ADD_TO_CART'){
+            setCartData(cartState.payload.cartlist)
+        }else if(cartState.type === 'REMOVE_FROM_CARTLIST'){
+            setCartData(cartState.payload)
+        }
+    },[cartState])
+
+    const increment = (id) => {
+        let newData = cartData.map((item) => 
+         item.id === id ? {...item , quantity: item.quantity + 1 }: item
```
Comment (git-0r) [inline src/pages/Cart/Cart.jsx:31]: suggestion: you can move these functions to cart reducer.

**Review by git-0r — APPROVED** [review]

File: src/backend/db/products.js [hunk 3]
```diff
@@ -7,213 +7,276 @@ import { v4 as uuid } from "uuid";
...
   // {
-  //   id: uuid(),
+  //   _id: uuid(),
   //   img : 'https://www.trackandtrail.in/sites/default/files/styles/banner/public/str2-brown.png?itok=2v7ZLnlq',
   //   desc : 'Roadeo STR2 29T Brown',
   //   price: '14,400',
   //   rating: '4.8',
   //   brand:'roadeo',
   //   type: 'city',
   //   gender:'male',
```
Comment (gouravmarch20) [inline src/backend/db/products.js:43]: remove comment

File: src/components/NavBar/Navbar.jsx [hunk 4]
```diff
@@ -3,8 +3,29 @@ import './Navbar.css';
...
+    let wishlistLength
+    if(wishlistState.wishlistState.type === 'ADD_TO_WISHLIST' || wishlistState.wishlistState.type === 'REMOVE_FROM_WISHLIST'){
+      wishlistLength = wishlistState.wishlistState.payload.length
+    }else if(wishlistState.wishlistState.type === 'ADD_TO_WISHLIST' || wishlistState.wishlistState.payload === 'none'){
+        wishlistLength = 0;
+    }
+
+    let cartState = useCartlist();
+    let cartlistLength;
+    if(cartState.cartState.type === 'ADD_TO_CART'){
```
Comment (gouravmarch20) [inline src/components/NavBar/Navbar.jsx:21]: can replace if-else block by : switch statement

File: src/context/Product-context.js [hunk 5]
```diff
@@ -10,123 +10,119 @@ const useProducts = () => useContext(ProductContext);
...
+     return products
   }
-
-  return ratedData;
   }
 
   // Brands Filter
-  const brandFilterData = () => {
-      let brandData = ratingFilterData();
+  const brandFilterData = (products) => {
```
Comment (gouravmarch20) [inline src/context/Product-context.js:64]: can create a utils folder: to write all repetitive logic ==> chance are high same methord will use in wishlist , cart , product page

**Review by gouravmarch20 — COMMENTED** [review]
few  major changes are required, breaking code into multiple moudule like utils , try to avoid to much if else block replace with swith case 

- good naming conventions.

**Pull request merged**